### PR TITLE
Fix player score not updating while game info screen is visible

### PR DESCRIPTION
**Pull request by obrakmann**
This fixes an inconsistency in the game info screen.  The team scores will update dynamically while the screen is visible, but the player scores will not. This fixes this oversight.

### CODE REVIEW

**Comment by pchote** [comment]
Lets get this in for the playtest.

File: OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs [hunk 1]
```diff
@@ -120,8 +120,7 @@ public GameInfoStatsLogic(Widget widget, World world, OrderManager orderManager,
 						item.Get<LabelWidget>("FACTION").GetText = () => pp.DisplayFaction.Name;
 					}
 
-					var experience = p.Second != null ? p.Second.Experience : 0;
-					item.Get<LabelWidget>("SCORE").GetText = () => experience.ToString();
+					item.Get<LabelWidget>("SCORE").GetText = () => (p.Second != null ? p.Second.Experience : 0).ToString();
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs]: If you don't mind the scope creep we could save some memory churn here by using `CachedTransform` to only allocate new strings when the experience actually changes.  It would be good to expand this to the other cells too.

Comment (obrakmann) [reply OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs]: erm... I'm probably doing it wrong, but it's not working:

```diff
--- a/OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs
+++ b/OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs
@@ -9,6 +9,7 @@
  */
 #endregion
 
+using System.Collections.Generic;
 using System.Drawing;
 using System.Linq;
 using OpenRA.Graphics;
@@ -77,8 +78,9 @@ public GameInfoStatsLogic(Widget widget, World world, OrderManager orderManager,
                                        var teamHeader = ScrollItemWidget.Setup(teamTemplate, () => true, () => { });
                                        teamHeader.Get<LabelWidget>("TEAM").GetText = () => t.Key == 0 ? "No Team" : "Team {0}".F(t.Key);
                                        var teamRating = teamHeader.Get<LabelWidget>("TEAM_SCORE");
-                                       teamRating.GetText = () => t.Sum(gg => gg.Second != null ? gg.Second.Experience : 0).ToString();
-
+                                       var teamScore = new CachedTransform<IGrouping<int, Pair<Player, PlayerStatistics>>, string>(p =>
+                                               p.Sum(gg => gg.Second != null ? gg.Second.Experience : 0).ToString());
+                                       teamRating.GetText = () => teamScore.Update(t);
                                        playerPanel.AddChild(teamHeader);
                                }
```

Comment (obrakmann) [reply OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs]: Updated

File: OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs [hunk 2]
```diff
@@ -77,7 +77,8 @@ public GameInfoStatsLogic(Widget widget, World world, OrderManager orderManager,
 					var teamHeader = ScrollItemWidget.Setup(teamTemplate, () => true, () => { });
 					teamHeader.Get<LabelWidget>("TEAM").GetText = () => t.Key == 0 ? "No Team" : "Team {0}".F(t.Key);
 					var teamRating = teamHeader.Get<LabelWidget>("TEAM_SCORE");
-					teamRating.GetText = () => t.Sum(gg => gg.Second != null ? gg.Second.Experience : 0).ToString();
+					var scoreCache = new CachedTransform<int, string>(s => s.ToString());
+					teamRating.GetText = () => scoreCache.Update(t.Sum(gg => gg.Second != null ? gg.Second.Experience : 0));
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs]: Can win a bit more here by caching and reusing the score enumerable:
```
var teamMemberScores = t.Select(tt => tt.Second)
	.Where(s => s != null)
	.ToArray()
	.Select(s => s.Experience);

	teamRating.GetText = () => scoreCache.Update(teamMemberScores.Sum());
```

Comment (obrakmann) [reply OpenRA.Mods.Common/Widgets/Logic/Ingame/GameInfoStatsLogic.cs]: done

**Review by pchote — APPROVED** [review]

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/bbb9eedfcffe70da57a2d22367333fd21d06f393...364fb548df91e0d2100e7f3b04c205b907ded907)
